public/app/features/dashboard/state/initDashboard.ts

```diff
@@ -206,6 +208,25 @@ export function initDashboard(args: InitDashboardArgs): ThunkResult<void> {
     }
 
     // yay we are done
+    if (dashDTO.dashboard.uid) {
+      dashboard.panels.forEach((panel) => {
+        panel.targets.forEach((target) => {
+          let dsSpecific = {};
+          if (target.datasource?.type === 'grafana-azure-monitor-datasource') {
+            dsSpecific = getAzureMonitorEvent(target as AzureMonitorQuery);
+          }
+          reportInteraction('grafana_dashboard_loaded', {
+            dashboard_id: dashDTO.dashboard.uid,
+            panel_id: panel.id,
+            hidden: !!target.hide,
+            datasource: target.datasource?.type,
+            grafana_version: config.buildInfo.version,
```

Comment:
Not sure how hard it would be:
Is it possible to also capture the http status code from the data source query and log? I understand this would mean you can log it only after the data source query returns, but that may also be interesting information to know how many times it failed.

Reply:
Also, would this have the trace id so we can correlate it with other logs?